### PR TITLE
[JENKINS-29150] - Fix the CronTabTest.testTimezone test

### PR DESCRIPTION
**Pull request by AbhyudayaSharma**
I cloned the Jenkins repository and tried to run `mvn compile`. The build failed with the following error:
```
[ERROR] Failures: 
[ERROR] CronTabTest.testTimezone:314 expected:<[[35, 5]6]> but was:<[[5, 2]6]>
```
I saw the code for the failing test and changed the time zone to match the time zone being tested against.

### Proposed changelog entries

* `Internal:` Fix `CronTabTest.testTimezone` JUnit test;

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention
@jenkinsci/code-reviewers

### CODE REVIEW

**Comment by AbhyudayaSharma** [comment]
Yes, it seems to be the same issue. I believe it has to do with India's time zone which is offset by 30 minutes (UTC+5:30).

File: core/src/test/java/hudson/scheduler/CronTabTest.java [hunk 1]
```diff
@@ -307,7 +308,9 @@ public int next(int n) {
         CronTabList tabs = CronTabList.create("TZ=Australia/Sydney\nH * * * *\nH * * * *", Hash.from("seed"));
         List<Integer> times = new ArrayList<Integer>();
         for (int i = 0; i < 60; i++) {
-            if (tabs.check(new GregorianCalendar(2013, 3, 3, 11, i, 0))) {
+            GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("Australia/Sydney"));
```
Comment (oleg-nenashev) [inline core/src/test/java/hudson/scheduler/CronTabTest.java]:
```suggestion
            GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
```

**Review by oleg-nenashev — COMMENTED** [review]
Yes, this is my understanding as well. I would suggest switching the test to UTC so that it does not look like that the TZ in the test itself has to be the same as in CRON

**Review by oleg-nenashev — APPROVED** [review]

**Comment by AbhyudayaSharma** [comment]
This seems to be another unrelated failure.

**Review by daniel-beck — APPROVED** [review]

**Review by batmat — APPROVED** [review]
Nice catch, thanks !

**Pull request merged**